### DISCUSSION TITLE
To RAID or not to RAID, that is my question.

### DISCUSSION THREAD

**Discussion by Br4v3St4rr** [longhorn/longhorn#3635] [Q&A]
Hi,

I've looked through the docs (possibly not in the correct place) and cant seem to find a mention of RAID.  I'm looking at a 3 node cluster with 5x 1TB SSDs.  Am I better presenting each set as a RAID array with 1 Replica or as a JBOD with 2?

Kind regards,

Ryan

**Comment by derekbit** [comment]
@Br4v3St4rr 
RAID aims to minimize the risk of data loss and node downtime if there is any disk failure locally.
On the other hand, LH is a distributed storage solution. The replicas can avoid the volume  error caused by the node failure, networking, other LH internal issues and so on.

Ideally, using RAID and replica2 is a better choice.

However, the choice highly depends on your environment. If the cluster is robust such as having enough nodes or you don't care about the node downtime, you can probably choose JBOD with 2.

**Reply by Br4v3St4rr** [reply]
@derekbit 
After some internal discussions and a double check of the docks I realised RAID + Replica 2(or more) is probably a better route.  LH cannot span volumes over disks so a volume would be constrained to a max capacity of the physical disk.  With a RAID array on each node you can overcome this while maintaining local disk failure protection. 

e.g.
5 x 1TB JBOD - Max volume size <1TB
5x 1TB RAID - Max volume size >1TB

Thanks for you response and guidance.

**Reply by derekbit** [reply]
Not really sure if the maximum size of LH volume in a JBOD is limited by the single disk size.

But, for the storage performance, the upper bound of a JBOD's performance is a single disk IO performance.
RAID 0 without any redundancy is a better alternative to JBOD, because the data striping of RAID can speed up the IO performance.

Ideally,
JBOD throughput = 1x disk throughput
RAID0 throughput = 5x disk throughput

RAID5 is better solution for both fault tolerance and performance.

**Reply by Br4v3St4rr** [reply]
Sorry maybe I miss understood https://github.com/longhorn/longhorn/issues/1061

Agreed RAID 5, or greater, seems to be the best for performance and fault tolerance.
